pkg/external-provider/metric_namer.go

```diff
@@ -0,0 +1,159 @@
+package provider
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	prom "github.com/directxman12/k8s-prometheus-adapter/pkg/client"
+	"github.com/directxman12/k8s-prometheus-adapter/pkg/config"
+	"github.com/directxman12/k8s-prometheus-adapter/pkg/naming"
+)
+
+var nsGroupResource = schema.GroupResource{Resource: "namespaces"}
+var groupNameSanitizer = strings.NewReplacer(".", "_", "-", "_")
+
+// MetricNamer knows how to convert Prometheus series names and label names to
+// metrics API resources, and vice-versa.  MetricNamers should be safe to access
+// concurrently.  Returned group-resources are "normalized" as per the
+// MetricInfo#Normalized method.  Group-resources passed as arguments must
+// themselves be normalized.
+type MetricNamer interface {
```

Comment:
This duplicates most of the code from `custom-provider/metric_namer.go`. Is it possible to merge those two implementations? I diff'ed locally and there seems to be some differences, albeit not many.

Reply:
So logically I feel like the metric namer should move to `naming`. Opinions ?

Reply:
`naming` sounds like a good choice :+1: